server2/Arcadia.Assistant/Arcadia.Assistant.Sharepoint/SharepointSynchronizator.cs

```diff
@@ -0,0 +1,171 @@
+﻿namespace Arcadia.Assistant.Sharepoint
+{
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Threading;
+    using System.Threading.Tasks;
+
+    using CalendarEvent;
+
+    using Employees.Contracts;
+
+    using ExternalStorages.Abstractions;
+
+    using Microsoft.Extensions.Logging;
+
+    using SickLeaves.Contracts;
+
+    using Vacations.Contracts;
+
+    using WorkHoursCredit.Contracts;
+
+    public sealed class SharepointSynchronizator
+    {
+        #region ctor
+
+        public SharepointSynchronizator(ISickLeaves sickLeaves, IVacations vacations, IWorkHoursCredit workouts, ISharepointDepartmentsCalendarsSettings departmentsCalendarsSettings, ILogger? logger)
+        {
+            this.sickLeaves = sickLeaves;
+            this.vacations = vacations;
+            this.workouts = workouts;
+            this.departmentsCalendarsSettings = departmentsCalendarsSettings;
+            this.logger = logger;
+        }
+
+        #endregion
+
+        #region publci interface
```

Comment:
Very "useful" regions and moreover with a typo :) Why do you ever need them?

Reply:
Fixed. Yes, i use it in my work.